File: android/src/main/java/net/xpece/android/util/XpLog.java
package net.xpece.android.util;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.view.View;

/**
 * @author Eugen on 11. 2. 2016.
 * @hide
 */
public final class XpLog {
    private XpLog() {}

    public static void logException(Throwable ex, Context context) {
        boolean debug = (context.getApplicationInfo().flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0;
        if (debug) ex.printStackTrace();
    }

    public static void logException(Throwable ex, View view) {
        boolean debug = (view.getContext().getApplicationInfo().flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0;
        if (debug) ex.printStackTrace();
    }
}

File: android/src/main/java/net/xpece/android/util/XpLog.kt
@file:JvmName("XpLog")

package net.xpece.android.util

import android.content.Context
import android.content.pm.ApplicationInfo
import android.view.View

fun Context.logException(ex: Throwable) {
    val debug = applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE != 0
    if (debug) ex.printStackTrace()
}

fun View.logException(ex: Throwable) {
    context.logException(ex)
}
